pkg/controller/alb-controller.go

```diff
@@ -107,13 +105,20 @@ func (ac *ALBController) syncALBs() {
 	for {
 		time.Sleep(ac.albSyncInterval)
 		logger.Debugf("ALB sync interval %s elapsed; assembling ingresses..", ac.albSyncInterval)
-		ac.ALBIngresses = albingresses.AssembleIngressesFromAWS(&albingresses.AssembleIngressesFromAWSOptions{
-			Recorder:    ac.recorder,
-			ClusterName: ac.clusterName,
-		})
+		ac.syncALBsWithAWS()
 	}
 }
 
+func (ac *ALBController) syncALBsWithAWS() {
+	albIngresses := albingresses.AssembleIngressesFromAWS(&albingresses.AssembleIngressesFromAWSOptions{
+		Recorder:    ac.recorder,
+		ClusterName: ac.clusterName,
+	})
+	ac.mutex.Lock()
```

Comment:
@matt-deboer - I'm thinking we should lock before attempting `AssembleIngressesFromAWS` otherwise an in flight reconcile could have changed AWS resources after we gathered them.

Thoughts?